### PR TITLE
Updated perfetto submodule to v46

### PR DESCRIPTION
**Pull request by ajanicijamd**
[SWDEV-479652](https://ontrack-internal.amd.com/browse/SWDEV-479652)
Some omnitrace tests were failing because we were using too old version of Perfetto (v28.0).
Updated Perfetto to v46.0.
Some source code and build configuration were necessary.

### CODE REVIEW

File: cmake/Perfetto.cmake
Comment (dgaliffiAMD) [inline cmake/Perfetto.cmake:1]: Do you mind sharing what warnings you saw? Thanks.

Comment (ajanicijamd) [reply cmake/Perfetto.cmake:1]: When build of Perfetto tries to compile protobuf:

```
[395/2478] compile ../../buildtools/protobuf/src/google/protobuf/message_lite.cc
FAILED: obj/buildtools/protobuf/src/google/protobuf/protobuf_lite.message_lite.o
/usr/bin/g++ -MMD -MF obj/buildtools/protobuf/src/google/protobuf/protobuf_lite.message_lite.o.d -DNDEBUG 
-DHAVE_PTHREAD=1 -DGOOGLE_PROTOBUF_NO_RTTI -DGOOGLE_PROTOBUF_NO_STATIC_INITIALIZER -DPERFETTO_IMPLEMENTATION 
-I../.. -I../../src/profiling/memory/include -I../../include -Igen/build_config -Igen -fstrict-aliasing 
-Wformat -g -fPIC -fstack-protector-strong -Werror -fno-omit-frame-pointer -mbmi -mbmi2 -mavx2 -mpopcnt 
-msse4.2 -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -fvisibility=hidden -O3 
-fdata-sections -ffunction-sections -Wno-stringop-overread -Wno-deprecated-declarations -isystem 
../../buildtools/protobuf/src -Wno-deprecated-declarations -isystem ../../buildtools/protobuf/src 
-fno-exceptions -fno-rtti -std=c++17  -static-libgcc -Wno-maybe-uninitialized  -static-libgcc 
-Wno-maybe-uninitialized  -c ../../buildtools/protobuf/src/google/protobuf/message_lite.cc -o 
obj/buildtools/protobuf/src/google/protobuf/protobuf_lite.message_lite.o
In file included from /usr/include/string.h:535,
                 from ../../buildtools/protobuf/src/google/protobuf/stubs/port.h:39,
                 from ../../buildtools/protobuf/src/google/protobuf/stubs/common.h:48,
                 from ../../buildtools/protobuf/src/google/protobuf/message_lite.h:46,
                 from ../../buildtools/protobuf/src/google/protobuf/message_lite.cc:36:
In function ‘void* memcpy(void*, const void*, size_t)’,
    inlined from ‘uint8_t* google::protobuf::io::EpsCopyOutputStream::WriteRaw(const void*, int, uint8_t*)’ 
	at ../../buildtools/protobuf/src/google/protobuf/io/coded_stream.h:684:16,
    inlined from ‘virtual uint8_t* google::protobuf::internal::ImplicitWeakMessage::_InternalSerialize(
	uint8_t*, google::protobuf::io::EpsCopyOutputStream*) const’ at 
	../../buildtools/protobuf/src/google/protobuf/implicit_weak_message.h:103:28,
    inlined from ‘bool google::protobuf::MessageLite::SerializePartialToZeroCopyStream(
	google::protobuf::io::ZeroCopyOutputStream*) const’ at 
	../../buildtools/protobuf/src/google/protobuf/message_lite.cc:411:30,
    inlined from ‘bool google::protobuf::MessageLite::SerializeToZeroCopyStream(
	google::protobuf::io::ZeroCopyOutputStream*) const’ at 
	../../buildtools/protobuf/src/google/protobuf/message_lite.cc:395:42:
/usr/include/x86_64-linux-gnu/bits/string_fortified.h:29:33: error: 
‘void* __builtin___memcpy_chk(void*, const void*, long unsigned int, long unsigned int)’ 
specified size between 18446744071562067968 and 18446744073709551615 exceeds maximum object size 
9223372036854775807 [-Werror=stringop-overflow=]
   29 |   return __builtin___memcpy_chk (__dest, __src, __len,
      |          ~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~
   30 |                                  __glibc_objsize0 (__dest));
      |                                  ~~~~~~~~~~~~~~~~~~~~~~~~~~

```

File: source/lib/core/perfetto.hpp
Comment (ajanicijamd) [inline source/lib/core/perfetto.hpp:1]: Error I was getting:

```
/home/aleks/Work/omnitrace/source/lib/core/perfetto.hpp:113:48: error: no matching function for call to ‘perfetto::CounterTrack::CounterTrack(<brace-enclosed initializer list>)’
  112 |     _track_data.emplace_back(::perfetto::CounterTrack{ _name->c_str() }
      |                              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  113 |                                  .set_unit_name(_unit_name)
      |                                  ~~~~~~~~~~~~~~^~~~~~~~~~~~
In file included from /home/aleks/Work/omnitrace/source/lib/core/perfetto.hpp:31,
                 from /home/aleks/Work/omnitrace/source/lib/omnitrace/library/cpu_freq.cpp:29:

```

File: source/lib/omnitrace/library/tracing.hpp
Comment (ajanicijamd) [inline source/lib/omnitrace/library/tracing.hpp:1]: Error I was getting:

```
/home/aleks/Work/omnitrace/source/lib/core/perfetto.hpp:113:48: error: no matching function for call to ‘perfetto::CounterTrack::CounterTrack(<brace-enclosed initializer list>)’
  112 |     _track_data.emplace_back(::perfetto::CounterTrack{ _name->c_str() }
      |                              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  113 |                                  .set_unit_name(_unit_name)
      |                                  ~~~~~~~~~~~~~~^~~~~~~~~~~~
In file included from /home/aleks/Work/omnitrace/source/lib/core/perfetto.hpp:31,
                 from /home/aleks/Work/omnitrace/source/lib/omnitrace/library/cpu_freq.cpp:29:

In file included from /home/aleks/Work/omnitrace/source/lib/core/perfetto.hpp:31,
                 from /home/aleks/Work/omnitrace/source/lib/omnitrace/library/tracing.hpp:31,
                 from /home/aleks/Work/omnitrace/source/lib/omnitrace/library.cpp:60:
/home/aleks/Work/omnitrace/source/lib/omnitrace/library/tracing.hpp: In lambda function:
/home/aleks/Work/omnitrace/source/lib/omnitrace/library/tracing.hpp:651:5: error: no matching function for call to ‘DecayEventNameType(perfetto::Track&)’
  651 |     TRACE_EVENT_INSTANT(trait::name<CategoryT>::value, _track, _ts,
      |     ^~~~~~~~~~~~~~~~~~~

```

**Review by dgaliffiAMD — APPROVED** [review]
Thank you. 
Unfortunately, it looks like we're still seeing those Perfetto CI failures.

**Pull request closed without merging**